process/sync/baseSync.go

```diff
@@ -1194,16 +1195,21 @@ func (boot *baseBootstrap) checkNeedsProofByNonce(
 	// if header does not exist
 	//		if it has a proof, request the header
 	//		if it does not have the proof, request both and decide when header is received if it truly needed the proof
-	_, errGetProof := boot.proofs.GetProofByNonce(nonce, boot.shardCoordinator.SelfId())
+	proof, errGetProof := boot.proofs.GetProofByNonce(nonce, boot.shardCoordinator.SelfId())
 	hasProof := errGetProof == nil
 	needsProof := !hasProof
 
 	if check.IfNil(header) {
 		return needsProof
```

Comment:
if proof will end up being requested by nonce (if not header at this point), it will be checked by hash later on on L1179